src/JsonReader.cs

```diff
@@ -189,6 +189,12 @@ public static IJsonReader<T> AsEnum<T>(this IJsonReader<string> reader, bool ign
         where TEnum : struct, Enum =>
         reader.Select(selector).Validate($"Invalid member for {typeof(TEnum)}.", IsEnumDefined);
 
+    public static IJsonReader<T?> Nullable<T>(this IJsonReader<T> reader, T? @null) where T : struct =>
+        Either(Null(@null), from v in reader select new T?(v));
+
+    public static IJsonReader<T?> Nullable<T>(this IJsonReader<T> reader, T? @null) where T : class =>
+        Either(Null(@null), from v in reader select v);
```

Comment:
How about defaulting the values for `@null`?

```suggestion
    public static IJsonReader<T?> Nullable<T>(this IJsonReader<T> reader, T? @null = default)
        where T : struct =>
        Null(@null).Or(from v in reader select (T?)v);

    public static IJsonReader<T?> Nullable<T>(this IJsonReader<T> reader, T? @null = default)
        where T : class =>
        Null(@null).Or(from v in reader select (T?)v);
```

The usage (e.g. in tests) will then just read natural.

While these are tricky signatures because it looks like they're overloading on constraints, which is not normally possible, they incidentally work because overloading actually happen on different resulting types for the second argument (`@null`): `Nullable<T>` and `T`. Since we are not yet version 1.0, I think it would be interesting to have these to experiment with them.

Also in this suggestion:

- Render the code identical in both cases.
- nit: Constraint on the next line
- nit: Use `Null` + `Or` instead (more compatible with potential future changes)

Reply:
Can we also rename these to `OrNull`, which speaks more to it accepts a JSON `null` than the resulting type.

Reply:
Thanks for your suggestion, agree with your observations and hence applied your suggestion. Renamed `Nullable` to `OrNull` with 36b521ba720b22df425c1cc154d8e9fea091d802.